robot_smach_states/src/robot_smach_states/manipulation/place.py

```diff
@@ -8,18 +9,20 @@
 # TU/e Robotics
 from ed.entity import Entity
 from robot_skills.arm.arms import PublicArm, GripperTypes
+from robot_skills.robot import Robot
 from .place_designator import EmptySpotDesignator
 from ..navigation.navigate_to_place import NavigateToPlace
 from ..utility import LockDesignator, ResolveArm, check_arm_requirements
 from ..util.designators import check_type
 from ..util.designators.utility import LockingDesignator
 from robot_smach_states.world_model.world_model import Inspect
+from ..util.designators.core import Designator
 
 
 class Put(smach.State):
     REQUIRED_ARM_PROPERTIES = {"required_gripper_types": [GripperTypes.GRASPING], }
 
-    def __init__(self, robot, item_to_place, placement_pose, arm):
+    def __init__(self, robot: Robot, item_to_place: Entity, placement_pose: FrameStamped, arm: PublicArm) -> None:
```

Comment:
Arm is a designator....

Reply:
All of the other variables are designators then according to the docs, should we change them too then?

Reply:
Yes, Designators can be typed. So that should be done. The `check_type` in the constructor should be replaced by `check_resolve_type`. As only designators are allowed.